src/Task3.java

```diff
@@ -0,0 +1,110 @@
+import java.util.Scanner;
+
+public class Task3 {
+    public static void main(String[] args) {
+        // Task 3
+        Scanner reader = new Scanner(System.in);
+
+        System.out.println("Task 3: ");
+        System.out.println("Menu: ");
+        System.out.println("1. Is string a palindrome");
+        System.out.println("2. Filter an Array");
+        System.out.println("3. Number is divisible by X.");
+        System.out.println("\nChoose option (1, 2 or 3): ");
+        int menuSelector = reader.nextInt();
+        reader.nextLine(); // "swallow" leftover newline
+
+        if (menuSelector == 1) {
```

Comment:
You have different logics within ifs each one should have invoked function which deals with those specific logics, it keeps the code clean and it keeps functions hold all logics that you need to use for that specific case.